src/ExampleApp.cpp

```diff
@@ -241,59 +253,81 @@ void ExampleApp::OnResume()
 	eegeoWorld.OnResume();
 }
 
-void ExampleApp::Update (float dt)
+void ExampleApp::Update (float dt, const float headTransform[])
 {
     Eegeo::EegeoWorld& eegeoWorld = World();
     
+    eegeoWorld.EarlyUpdate(dt);
+    Eegeo::Camera::CameraState cameraState(m_pExampleController->GetCurrentCameraState());
+
     m_pCameraTouchController->Update(dt);
 
-	eegeoWorld.EarlyUpdate(dt);
-    
     m_pExampleController->EarlyUpdate(dt);
-    
-    m_pInteriorModule->Update(dt);
-    Eegeo::Camera::CameraState cameraState(m_pExampleController->GetCurrentCameraState());
-    Eegeo::Streaming::IStreamingVolume& streamingVolume(m_pExampleController->GetCurrentStreamingVolume());
-    
-    Eegeo::EegeoUpdateParameters updateParameters(dt,
-                                                  cameraState.LocationEcef(),
-                                                  cameraState.InterestPointEcef(),
-                                                  cameraState.ViewMatrix(),
-                                                  cameraState.ProjectionMatrix(),
-                                                  streamingVolume,
-                                                  m_screenPropertiesProvider.GetScreenProperties());
-    
-	eegeoWorld.Update(updateParameters);
-    
-    m_pExampleController->Update(dt);
-    
+
+    if(m_pExampleController->IsVRExample())
+    {
+        m_pExampleController->SetVRCameraState(headTransform);
+        m_pVRCardboardController->Update(dt, cameraState, eegeoWorld, m_vrModeTracker);
+    }
+    else
+    {
+        AppUpdate(dt, cameraState, eegeoWorld);
+    }
+
+    if(m_pLoadingScreen==NULL || m_pLoadingScreen->IsDismissed())
+    {
+        m_pExampleController->Update(dt);
+    }
+
     UpdateLoadingScreen(dt);
 }
 
 void ExampleApp::Draw (float dt)
 {
-    m_pExampleController->PreWorldDraw();
-    
     Eegeo::EegeoWorld& eegeoWorld = World();
-    
-    Eegeo::Camera::CameraState cameraState(m_pExampleController->GetCurrentCameraState());
-    
-    Eegeo::EegeoDrawParameters drawParameters(cameraState.LocationEcef(),
-                                              cameraState.InterestPointEcef(),
-                                              cameraState.ViewMatrix(),
-                                              cameraState.ProjectionMatrix(),
-                                              m_screenPropertiesProvider.GetScreenProperties());
-    
-    eegeoWorld.Draw(drawParameters);
-    
-    m_pExampleController->Draw();
-    
-    if (m_pLoadingScreen != NULL)
+
+    if(m_pExampleController->IsVRExample())
```

Comment:
As above. Don't think ExampleApp should be making decisions based on the "VR'ness" of the current example.